vault/resource_terraform_cloud_secret_role.go

```diff
@@ -0,0 +1,229 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	terraformCloudSecretBackendRoleBackendFromPathRegex = regexp.MustCompile("^(.+)/role/.+$")
+	terraformCloudSecretBackendRoleNameFromPathRegex    = regexp.MustCompile("^.+/role/(.+$)")
+)
+
+func terraformCloudSecretRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: terraformCloudSecretRoleWrite,
+		Read:   terraformCloudSecretRoleRead,
+		Update: terraformCloudSecretRoleWrite,
+		Delete: terraformCloudSecretRoleDelete,
+		Exists: terraformCloudSecretRoleExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of an existing role against which to create this Terraform Cloud credential",
+			},
+			"path": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The path of the Terraform Cloud Secret Backend the role belongs to.",
+				Deprecated:    "use `backend` instead",
```

Comment:
Since these are new resources, we should probably not include deprecated fields in the schema. Is there a reason we need them here? Possibly this is copy-paste or generated ?

Reply:
Yeah, that was copy-pasta'd. I debated removing it, but figured I would leave it in for the sake of people unwilling to change? But, probably not the best call overall.

Reply:
Removed